src/map.rs

```diff
@@ -235,13 +240,7 @@ impl<'a, K: Ord + 'a, V: 'a> IntoIterator for &'a ManagedMap<'a, K, V> {
     type IntoIter = Iter<'a, K, V>;
```

Comment:
This is not the [signature of `impl IntoIterator for BTreeMap`](https://doc.rust-lang.org/nightly/std/collections/btree_map/struct.BTreeMap.html#impl-IntoIterator-1). Specifically, you're returning borrowed items and you need to return owned ones.

Why do you need `.into_iter()` for IGMP anyway?

Reply:
> This is not the signature of impl IntoIterator for BTreeMap. Specifically, you're returning borrowed items and you need to return owned ones.

Is that significant? I thought leaving out cloning is beneficial to performance. To me it doesn't matter much as for IGMP I'm storing only 4-byte `Ipv4Address` in there.

> Why do you need .into_iter() for IGMP anyway?

For iterating through all `ipv4_mcast_groups` to find the next group to report to.

Actually, this case needs only a "set" which is why I'm using `()` as the map's values. Should we create a `ManagedSet` interface on top of `ManagedMap`?

Reply:
No cloning. If you look at the impl signature you'll see that it doesn't require `K: Clone` or `V: Clone`. That's not how IntoIterator works--it consumes the original map in the process, so it can move original elements out of the map.

> For iterating through all ipv4_mcast_groups to find the next group to report to.

What's wrong with `.iter()` exactly?

Reply:
Just to be clear: drop `into_iter()`, keep `iter()`?

Reply:
Sure. I mean, there's nothing inherently wrong with a correct implementation of `.into_iter()`, but the current one isn't.

Reply:
Ok, removed!